src/Tickets/Commerce/Order.php

```diff
@@ -514,6 +532,92 @@ public function create( Gateway_Interface $gateway, $args ) {
 		return tec_tc_orders()->set_args( $args )->create();
 	}
 
+	/**
+	 * Filters the values and creates a new Order with Tickets Commerce or updates an existing one.
+	 *
+	 * @since TBD
+	 *
+	 * @param Gateway_Interface $gateway           The gateway to use to create the order.
+	 * @param array             $args              The arguments to create the order.
+	 * @param ?int              $existing_order_id The ID of an existing order to update.
+	 *
+	 * @return false|WP_Post
+	 */
+	public function upsert( Gateway_Interface $gateway, array $args, ?int $existing_order_id = null ) {
```

Comment:
```suggestion
	public function upsert( Gateway_Interface $gateway, array $args ) {
```
Typically the upsert determines whether the object in question has the fields required to update versus insert. In this case that would be inferred from the `$args` param. I wouldn't leave the logic outside the params, it moves some of the logic for the Order details outside the upsert, and isn't very obvious to me that it would be required me to pass that optional order ID param. 

In other words, if I don't always pass that optional order ID it will always create, thus no longer being an `upsert` .

I think it should be moved inside the method, like...
```
$id = $args['id'] ?? 0;
if($id > 0) {
// Do insert 
} else {
// Do update 
}
```